File: app/src/main/java/io/github/mohamedisoliman/mvi/mymvi/ViewModelFactory.java
package io.github.mohamedisoliman.mvi.mymvi;

import android.arch.lifecycle.ViewModel;
import android.arch.lifecycle.ViewModelProvider;

/**
 * Created by Mohamed Ibrahim on 9/27/18.
 */
public class ViewModelFactory extends ViewModelProvider.NewInstanceFactory {
    private MviLoginView mParam;

    public ViewModelFactory(MviLoginView param) {
        mParam = param;
    }


    @Override
    public <T extends ViewModel> T create(Class<T> modelClass) {
        return (T) new LoginViewModel(mParam);
    }
}

File: app/src/main/java/io/github/mohamedisoliman/mvi/ViewModelFactory.kt
package io.github.mohamedisoliman.mvi

import android.arch.lifecycle.ViewModel
import android.arch.lifecycle.ViewModelProvider
import io.github.mohamedisoliman.mvi.login.LoginViewModel
import io.github.mohamedisoliman.mvi.login.MviLoginView

/**
 * Created by Mohamed Ibrahim on 9/27/18.
 */
class ViewModelFactory(private val mParam: MviLoginView) : ViewModelProvider.NewInstanceFactory() {

    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        return LoginViewModel(mParam) as T
    }
}
